inception-imls-core/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/core/classifier/Classifier.java

```diff
@@ -157,12 +164,15 @@ public Classifier(ClassifierConfiguration<C> conf)
                         predictionsWithMergedLabels.add(mergedAo);
                         id++;
                     }
-                    j++;
                 }
                 i++;
             }
         }
         
         return predictionsWithMergedLabels;
     }
+
+    public abstract void setUser(User user);
+
+    public abstract void setProject(Project project);
```

Comment:
You can reduce the number of files affected by adding user and project as fields to the abstract class classifier and fully implement setters and getters. In your NamedEntityLinkingClassifier, you can then just call the getter.